docs/design/conan.md

```diff
@@ -0,0 +1,373 @@
+  $ ls $CONAN_HOME/p/b/zlib3de6e91b5a12b/p
+  conaninfo.txt  conanmanifest.txt  include  lib  licenses
+  $ ls $CONAN_HOME/p/b/zlib3de6e91b5a12b/p/lib
+  libz.a
+  $ ls $CONAN_HOME/p/b/zlib3de6e91b5a12b/p/include/
+  zconf.h  zlib.h
+  # Sources are in:
+  $ ls $CONAN_HOME/p/zlib4f058de3f8852/s
+  patches  src
+  $ ls $CONAN_HOME/p/zlib4f058de3f8852/s/src
+  adler32.c       deflate.c  gzwrite.c   inftrees.h    qnx            zconf.h          zlib.pc.in
+  amiga           deflate.h  INDEX       LICENSE       README         zconf.h.cmakein  zutil.c
+  ...
+  ```
+</details>
+
+One possible solution to the problem of a build that happens during an install
+could be monkey-patching of `build()` method in any recipe that is to be
+downloaded.  The question of how to do this efficiently remains open, also some
+recipes download their sources as part of `build()`.
+
+The process of downloading sources for a recipe is supposed to be controlled by
+`source()` method and a corresponding subcommand. Judging from current state of
+Conan recipes on Github (1849 entires) all recipes have conanfile.py and most
+recipes define a separate `source()` method, just 29 do not do that. For example cmake
+just downloads the sources during a build.
+
+On the other hand 1839 of 1849 contain `conandata.yml` which could be used for
+collecting and injecting sources into a cache. A notable exception is `xorg`
+which apparently contains just a list of libraries to be installed with a
+package manager (i.e. without usage of any Conan-native tooling).
+
+`download` does not seem to be doing anything helpful at all: it downloads just
+a binary form without any transitive dependencies. There does not seem to be a
+way to make it download sources as well.
+
+  ```bash
+    $ conan download -r=conancenter zlib/1.2.13
+    Connecting to remote 'conancenter' anonymously
+    Skip recipe zlib/1.2.13#9df41c65e2c2b6ef47633dc32e0b699a download, already in cache
+    Downloading package 'zlib/1.2.13#9df41c65e2c2b6ef47633dc32e0b699a:41ad450120fdab2266b1185a967d298f7ae5259
+    5#58e0515ae415ebd08d63ea0db6fd1761'
+    ...
+    $ ls $CONAN_HOME/p
+    cache.sqlite3      zlib53546468985f4  zlib719177b87d205  zliba265d66faee39
+    zlib17bfee322e8d5  zlib561027700ae6e  zlib92b0cd6d35aaa  zlibc6a4b1fe33208
+    zlib4f058de3f8852  zlib5e2678cdd172f  zlib94fda36d28e71  zlibf12c65c363f25
+    $ ls $CONAN_HOME/p/zlib17bfee322e8d5/
+    d/ p/
+    $ ls $CONAN_HOME/p/zlib17bfee322e8d5/p
+    conaninfo.txt  conanmanifest.txt  include  lib  license
+  ```
+
+There are very few environment variables which affect Conan, the one useful is
+CONAN_HOME. All other settings should be done via profile or global.conf residing
+in CONAN_HOME.
+
+Another option that is present in Conan, but which does not seem to be used in the
+wild is package lock. A package could be locked with `conan lock create <path>`.
+That would create a lock file similar to one below:
```

Comment:
This means that the whole Conan backend would be strictly dependent on the `--permissive` mode just because lockfiles are not common.

Reply:
resolving as we're not implementing this any time soon.